crates/unftp-sbe-gcs/src/lib.rs

```diff
@@ -451,3 +527,46 @@ fn result_based_on_http_status<T>(status: StatusCode, ok_val: T) -> Result<T, Er
     }
     Ok(ok_val)
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[tokio::test]
+    async fn cached_token() {
+        let cache: CachedToken = Default::default();
+
+        assert_eq!(cache.get().await, None);
+
+        cache
+            .set(Token {
+                value: "the_value".to_string(),
+                expires_at: None,
+            })
+            .await;
+        assert_eq!(cache.get().await, None);
```

Comment:
to me this seems a bit weird. If i don't have an expiration then why is my token invalid?

Reply:
If we cant have none as expiration date we should just rule that out at the type level, avoid the confusing semantics

Reply:
Expiry information is not required in an OAuth2 access token. I took the 'safer' way and not cache the access token indefinitely, but I'm wondering whether that's the right way to go.

Come to think of it, this PR does not address cache invalidation yet. Should that be part of this PR or something we can think of later @hannesdejager?

Reply:
Separate issue, let's get this one shipped first.